src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -1306,7 +1313,44 @@ public virtual string Labels
         public virtual string ProxyUriPath { get { return _localConfiguration.service.proxy.uriPath; } }
         public virtual int ProxyPort { get { return _localConfiguration.service.proxy.port; } }
         public virtual string ProxyUsername { get { return _localConfiguration.service.proxy.user; } }
-        public virtual string ProxyPassword { get { return _localConfiguration.service.proxy.password; } }
+
+        private string _obscuringKey;
+        public string ObscuringKey
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_obscuringKey))
+                {
+                    _obscuringKey = EnvironmentOverrides(_localConfiguration.service.obscuringKey, "OBSCURING_KEY");
+                }
+
+                return _obscuringKey;
+            }
+        }
+
+        private string _proxyPassword;
+        public virtual string ProxyPassword
+        {
+            get
+            {
+                if (!_proxyPasswordEvaluated)
+                {
+                    if (_hasObscuringKey && _hasEncryptedPassword)
```

Comment:
I don't think we need _hasObscuringKey and _hasEncryptedPassword.  It seems like the only time we use these is in this function, so we should just evaluate them here.